pkgs/top-level/all-packages.nix

```diff
@@ -11722,27 +11727,28 @@ with pkgs;
   jdk11_headless = openjdk11_headless;
 
   /* Latest JDK */
-  openjdk16 =
+  openjdk17 =
     if stdenv.isDarwin then
       callPackage ../development/compilers/openjdk/darwin { }
     else
-      callPackage ../development/compilers/openjdk/16.nix {
+      callPackage ../development/compilers/openjdk/17.nix {
         openjfx = openjfx15;
         inherit (gnome2) GConf gnome_vfs;
       };
 
-  openjdk16_headless =
+  openjdk17_headless =
     if stdenv.isDarwin then
-      openjdk16
+      openjdk17
     else
-      openjdk16.override { headless = true; };
+      openjdk17.override { headless = true; };
 
-  jdk16 = openjdk16;
-  jdk16_headless = openjdk16_headless;
+  jdk17 = openjdk17;
+  jdk17_headless = openjdk17_headless;
 
   /* default JDK */
 
-  jdk = jdk16;
+  jdk-lts = jdk17;
```

Comment:
I don't think introducing `jdk-lts` attribute is a good idea. If an application doesn't use internal APIs and doesn't parse its own bytecode then using `jdk` will work fine. If that's not the case then updating to newest JDK is going to break stuff. If an user wants an LTS, I think having to specify a version makes sense.

Reply:
I agree, the existing `jdk` alias itself seems good enough